app/helpers/hyku_addons/cross_tenant_shared_search_helper.rb

```diff
@@ -0,0 +1,38 @@
+# frozen_string_literal: true
+
+# overriding flipflop
+module HykuAddons
+  module CrossTenantSharedSearchHelper
+    def redirect_to_shared_search_page
+      cname = current_account.parent&.cname
+      return unless cname.present?
+
+      local_url = "#{request.protocol}#{cname}:#{request.port}/catalog?utf8=✓&locale=en&search_field=all_fields&q="
+      production_url = "#{request.protocol}#{cname}/catalog?utf8=✓&locale=en&search_field=all_fields&q="
+
+      request.host.include?('hyku.docker') ? local_url : production_url
+    end
+
+    def tenants_not_in_search(account)
+      @list_for_unsaved_record = Account.not_cross_search_tenants_new_list - account.children
+      @list_without_record_under_edit = Account.not_cross_search_tenants_edit_list(account.id) - account.children
+      account.new_record? ? @list_for_unsaved_record : @list_without_record_under_edit
```

Comment:
I am happy to move it to the model

Reply:
Just a quick update, I have moved the code above to the model,.  tried applying the code  changes suggested above but it does not work the old method.

1  The assumption in your code changes above assumes the scope expects many ids instead of one id.  

2. Expecting many ids does take into account the scope chaining that fetches some of the ids we are now passing as part of the items_to_exclude.

I agree that the method could be simply and will be happy to take a look within the week hopefully.